newIDE/app/src/ProjectCreation/ResolutionOptions.js

```diff
@@ -0,0 +1,270 @@
+
+// Styles to give the impression of pressing an element.
+const useStylesForButtonBase = (selected: boolean) =>
+  makeStyles(theme =>
+    createStyles({
+      root: {
+        outline: selected
+          ? `2px solid ${theme.palette.secondary.dark}`
+          : `1px solid ${theme.palette.text.disabled}`,
+        '&:focus': {
+          backgroundColor: theme.palette.action.hover,
+        },
+        '&:hover': {
+          backgroundColor: theme.palette.action.hover,
+        },
+      },
+    })
+  )();
+
+type Props = {|
+  children: React.Node,
+  onClick: () => void,
+  selected: boolean,
+  disabled?: boolean,
+|};
+
+const ResolutionOptionButton = ({
+  children,
+  onClick,
+  selected,
+  disabled,
+}: Props) => {
+  const classes = useStylesForButtonBase(selected);
+
+  return (
+    <ButtonBase
+      onClick={onClick}
+      elevation={2}
+      style={{
+        ...styles.buttonBase,
+      }}
+      classes={classes}
+      tabIndex={0}
+      onKeyPress={(event: SyntheticKeyboardEvent<HTMLLIElement>): void => {
+        if (shouldValidate(event) && !selected) {
+          onClick();
+        }
+      }}
+      disableTouchRipple={selected} // Avoid ripple effect even if already selected.
+      disabled={disabled}
+    >
+      <div style={styles.contentWrapper}>{children}</div>
+    </ButtonBase>
+  );
+};
+
+const ResolutionOptions = ({
+  selectedOption,
+  onClick,
+  disabled,
+  customWidth,
+  customHeight,
+  onCustomWidthChange,
+  onCustomHeightChange,
+}: {|
+  selectedOption: string,
+  onClick: ResolutionOption => void,
+  disabled?: boolean,
+  customWidth: ?number,
+  customHeight: ?number,
+  onCustomWidthChange: (?number) => void,
+  onCustomHeightChange: (?number) => void,
+|}) => {
+  return (
+    <div style={styles.optionsContainer}>
+      {Object.keys(resolutionOptions).map((key, index) => {
+        const { width, height, label, icon } = resolutionOptions[key];
+        return (
+          <Column expand noMargin key={key}>
+            <ResolutionOptionButton
+              onClick={() => onClick(key)}
+              selected={selectedOption === key}
+              disabled={disabled}
+            >
+              <Column
+                expand
+                alignItems="center"
+                justifyContent="center"
+                noMargin
+              >
+                {icon}
+                <Spacer />
+                <Column noMargin>
+                  <Text size="body2" noMargin>
+                    {label}
+                  </Text>
+                  {width && height ? (
+                    <Text size="body-small" noMargin color="secondary">
+                      {width}x{height}
+                    </Text>
+                  ) : (
+                    <Line noMargin alignItems="center" justifyContent="center">
+                      <Text size="body-small" noMargin color="secondary">
+                        W
+                      </Text>
+                      <TextField
```

Comment:
Is it possible to use a SemiControlledTextfield? If you have 800 entered but you want 900, if you put your caret such as `8|00` and erase the 8 to put a 9 instead, everything disappears